cli.py

```diff
@@ -0,0 +1,166 @@
+import sys
+import argparse
+import bech32m
+import enc_dec
+
+
+def user_interface():
+    """
+    provides cli and stdin/stdout options to user for encoding/decoding
+    :return: mode, hrp, input_string, infile, input_format, output_format, outfile
+    """
+
+    if len(sys.argv) > 1:
+        my_parser = argparse.ArgumentParser(description='Bech32m Encoder/Decoder')
+        my_parser.add_argument('m', type=str, help='encode/decode', choices=['enc', 'dec'])
+        my_parser.add_argument('hrp', type=str, help='human readable part[ASCII]')
+        group = my_parser.add_mutually_exclusive_group(required=True)
+        group.add_argument('-s', type=str, help='input string to be encoded/decoded')
+        group.add_argument('-inf', type=str, help='input file to read string to be encoded/decoded')
+        my_parser.add_argument('-iform', type=str, choices=['hex', 'bin', 'b64'], help='input format')
+        my_parser.add_argument('-oform', type=str, choices=['hex', 'bin', 'b64'], help='output format')
+        my_parser.add_argument('-of', type=str, help='output file path to write encoded/decoded string')
+        args = my_parser.parse_args()
+
+        if args.s is not None and args.iform == "bin":
+            args_s = args.s
+            args_str = args_s.encode()
+        else:
+            args_str = args.s
+
+        return args.m, args.hrp, args_str, args.inf, args.iform, args.oform, args.of
+
+    if len(sys.argv) == 1:
+        print("Default Interface Stdin Selected")
+
+        #Enter Correct Mode
+        mode = ""
+        while mode not in ["enc", "dec"]:
+            mode = input("Enter the mode [\"enc\" or \"dec\"]:")
+            if mode not in ["enc", "dec"]:
+                print("Enter Valid Mode [\"enc\" or \"dec\"]")
+
+        #Enter and Validate HRP
+        hrp = input("Enter HRP[valid ascii]:")
+        while not enc_dec.validate_hrp(hrp):
+            hrp = input("Enter HRP[valid ascii]:")
+
+        #Enter and validate input format
+        input_format = input("Enter Input Format[b64/bin/hex]"
+                             "(Press Enter for Default[\"hex\" for \"enc\" and \"bech32m\" for \"dec\"]):")
+        while input_format not in ["b64", "bin", "hex", ""]:
+            input_format = input("Enter Valid Input Format[b64/bin/hex]"
+                                 "(Press Enter for Default[\"hex\" for \"enc\" and \"bech32m\" for \"dec\"]):")
```

Comment:
issue (maintainability): There are repeated prompts in the code, I suggest either using a special function for getting user inputs, or at least parametrizing the messages and changing them only in one place.